pgrx/src/shmem.rs

```diff
@@ -184,12 +187,11 @@ impl PgSharedMem {
 
             lock.attach(fv_shmem);
             pg_sys::LWLockRelease(addin_shmem_init_lock);
-        }
     }
 
     /// Must be run from the shared memory init hook, use for rust atomics behind `PgAtomic`
-    pub fn shmem_init_atomic<T: atomic_traits::Atomic + Default>(atomic: &PgAtomic<T>) {
-        unsafe {
+    /// SAFETY: Can only be called from Postgres its shared memory init hook
+    pub unsafe fn shmem_init_atomic<T: atomic_traits::Atomic + Default>(atomic: &PgAtomic<T>) {
```

Comment:
We can put this unsafe block back, and in so doing not have to reformat everything.
```suggestion
    /// # Safety
    /// Must only be called from Postgres its shared memory init hook
    pub unsafe fn shmem_init_atomic<T: atomic_traits::Atomic + Default>(atomic: &PgAtomic<T>) {
        unsafe {
```